src/neuroconv/datainterfaces/ophys/scanimage/scanimageimaginginterfaces.py

```diff
@@ -1,88 +1,281 @@
-        """
-        Get the source schema for the ScanImage imaging interface.
-
-        Returns
-        -------
-        dict
-            The schema dictionary containing input parameters and descriptions
-            for initializing the ScanImage interface.
-        """
-        source_schema = super().get_source_schema()
-        source_schema["properties"]["file_path"]["description"] = "Path to Tiff file."
-        return source_schema
-
-    @validate_call
-    def __new__(
-        cls,
-        file_path: FilePath,
-        channel_name: str | None = None,
+    def __init__(
+        self,
+        file_path: Optional[FilePath] = None,
+        channel_name: Optional[str] = None,
+        slice_sample: Optional[int] = None,
+        plane_index: Optional[int] = None,
+        file_paths: Optional[list[str]] = None,
+        interleave_slice_samples: Optional[bool] = None,
         plane_name: str | None = None,
         fallback_sampling_frequency: float | None = None,
         verbose: bool = False,
     ):
-        from roiextractors.extractors.tiffimagingextractors.scanimagetiff_utils import (
-            extract_extra_metadata,
-            parse_metadata,
-        )
+        """
+        Initialize the ScanImage Imaging Interface.
 
-        image_metadata = extract_extra_metadata(file_path=file_path)
-        version = get_scanimage_major_version(scanimage_metadata=image_metadata)
-        if version == "3.8":
-            return ScanImageLegacyImagingInterface(
-                file_path=file_path,
-                fallback_sampling_frequency=fallback_sampling_frequency,
-                verbose=verbose,
+        Parameters
+        ----------
+        file_path : PathType
+            Path to the TIFF file. If this is part of a multi-file series, this should be the first file.
+        channel_name : str, optional
+            Name of the channel to extract. If None and multiple channels are available, the first channel will be used.
+        file_paths : List[PathType], optional
+            List of file paths to use. If provided, this overrides the automatic
+            file detection heuristics. Use this if automatic detection does not work correctly and you know
+            exactly which files should be included.  The file paths should be provided in an order that
+            reflects the temporal order of the frames in the dataset.
+        interleave_slice_samples : bool, default: True
+            Controls whether to interleave all slice samples as separate time points when frames_per_slice > 1:
+            - If True: Interleaves all slice samples as separate time points, increasing the effective
+              number of samples by frames_per_slice. This treats each slice_sample as a distinct sample.
+            - If False (default): Requires a specific slice_sample to be provided when frames_per_slice > 1.
+            - This parameter has no effect when frames_per_slice = 1 or when slice_sample is provided.
```

Comment:
I'm not sure I get this how this works. Also `slice_sample` is missing from docstring

Reply:
Ok, solved in a discussion and explained it. We should add more through documentation in the second PR.

Reply:
Done